app/controllers/SubmitPoaController.scala

```diff
@@ -61,7 +70,147 @@ class SubmitPoaController @Inject()(cc: MessagesControllerComponents,
     }
   }
 
-  def extractNino(json: JsValue): Option[String] = {
+  private def overwriteTotalAmount(nino: String, json: JsValue): Unit = {
+    (extractPoAAmount(json), extractTaxYear(json)) match {
+      case (Some(amount), Some(taxYearString)) => {
+        TaxYear.createTaxYearGivenTaxYearRange(taxYearString) match {
+          case Some(taxYear) => {
+            val financialUrl = getFinancialDetailsUrl(nino, taxYear)
+            val financialDetailsResponse = dataRepository.find(equal("_id", financialUrl))
+            financialDetailsResponse.map {
+              case Some(value) => value.response match {
+                case Some(response) =>
+                  //extract chargeReference for 1554 data changes
+                  val chargeRef = extractChargeRef(response).getOrElse("None")
```

Comment:
I think you mean ?
`val chargeRef = extractChargeRef(response).getOrElse(None) `